src/psyclone/tests/psyir/nodes/array_mixin_test.py

```diff
@@ -395,18 +448,68 @@ def test_member_get_lbound_expression(fortran_writer):
     ssym = DataSymbol("var", stypedef2)
     sref = StructureReference.create(ssym,
                                      ["grid",
-                                      ("map", [two.copy(), two.copy()])])
-    assert sref.member.member.get_lbound_expression(0) == one
-    assert sref.member.member.get_lbound_expression(1) == two
+                                      ("map", [_TWO.copy(), _TWO.copy()])])
+    assert sref.member.member._get_bound_expression(0, "lower") == _ONE
+    assert sref.member.member._get_bound_expression(1, "lower") == _TWO
     sref2 = StructureReference.create(
-        ssym, [("subgrids", [two.copy(), two.copy()]),
-               ("map", [two.copy(), two.copy()])])
-    assert sref2.member.get_lbound_expression(1) == two
-    assert sref2.member.member.get_lbound_expression(1) == two
+        ssym, [("subgrids", [_TWO.copy(), _TWO.copy()]),
+               ("map", [_TWO.copy(), _TWO.copy()])])
+    assert sref2.member._get_bound_expression(1, "lower") == _TWO
+    assert sref2.member.member._get_bound_expression(1, "lower") == _TWO
 
+    # Check that get_lbound_expression gives the same result
+    assert (sref2.member.member._get_bound_expression(1, "lower") ==
+            sref2.member.member.get_lbound_expression(1))
+
+    # Tests for upper argument
+    # First, test when we don't have type information.
```

Comment:
Please could you merge this with the earlier code that tests for lbound. e.g. You could move L469-470 to L425 (and remove L466-467)?

Reply:
Yes - cleaned this up.